packages/plugin-ext/src/plugin/documents.ts

```diff
@@ -255,10 +255,10 @@ export class DocumentsExtImpl implements DocumentsExt {
             if (options.selection) {
                 const { start, end } = options.selection;
                 selection = {
-                    startLineNumber: start.line,
-                    startColumn: start.character,
-                    endLineNumber: end.line,
-                    endColumn: end.character
+                    startLineNumber: start.line + 1,
+                    startColumn: start.character + 1,
+                    endLineNumber: end.line + 1,
+                    endColumn: end.character + 1
```

Comment:
It looks like you have to add these `+1` because `theia` has `-1` [here](https://github.com/eclipse-theia/theia/blob/6125352297da5c0431819ac172e69830670c0c78/packages/plugin-ext/src/main/browser/documents-main.ts#L242).

So, maybe it's true way to remove those `-1` instead of adding these `+1`.
Could you check it?

Reply:
Both are, unfortunately, correct. The problem is we have two different types named `Range` and they have different indexing. 

1-based indexing for this type:
https://github.com/eclipse-theia/theia/blob/12f3389a6fcfed7143c1dc5aef5a2c507d7bc38f/packages/plugin-ext/src/common/plugin-api-rpc-model.ts#L53-L70

And 0-based indexing for this type: 
https://github.com/eclipse-theia/theia/blob/12f3389a6fcfed7143c1dc5aef5a2c507d7bc38f/packages/plugin/src/theia.d.ts#L343-L352
because this type has 0-based indexing:
https://github.com/eclipse-theia/theia/blob/12f3389a6fcfed7143c1dc5aef5a2c507d7bc38f/packages/plugin/src/theia.d.ts#L230-L243

It seems like we translate between those two types of `Range`s more often than necessary, but every time we do, we have to be sure to add or subtract one, depending on the direction.

As I say in my PR description:
> Fixes a bunch of off-by-one translation errors between RPC Model Ranges and VSCode Ranges. **_That is an absolute mess_** that deserves separate treatment. The translation is implemented separately at least 5 times.

There are functions specialized in the translation between the two types, but they aren't used everywhere, and it didn't seem like this PR was the right place to make that refactoring everywhere where the translation is currently implemented.

Reply:
@RomanNikitenko, I've added an issue for this here: https://github.com/eclipse-theia/theia/issues/9739 .

Reply:
> The problem is we have two different types named Range and they have different indexing.

I see...
thanks!